pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/BrokersBase.java

```diff
@@ -290,21 +295,30 @@ public void isReady(@Suspended AsyncResponse asyncResponse) {
         }
     }
 
-    @GET
-    @Path("/health")
-    @ApiOperation(value = "Run a healthcheck against the broker")
-    @ApiResponses(value = {
-        @ApiResponse(code = 200, message = "Everything is OK"),
-        @ApiResponse(code = 403, message = "Don't have admin permission"),
-        @ApiResponse(code = 404, message = "Cluster doesn't exist"),
-        @ApiResponse(code = 500, message = "Internal server error")})
-    public void healthcheck(@Suspended AsyncResponse asyncResponse) throws Exception {
-        validateSuperUserAccess();
-        String heartbeatNamespace = NamespaceService.getHeartbeatNamespace(
-                pulsar().getAdvertisedAddress(), pulsar().getConfiguration());
-        String topic = String.format("persistent://%s/healthcheck", heartbeatNamespace);
+    protected void healthCheck(final AsyncResponse asyncResponse,
+                               final TopicVersion topicVersion) throws PulsarServerException {
+
+        String topic = null;
+        PulsarClient client = null;
+        try {
+            validateSuperUserAccess();
+
+            if (topicVersion == TopicVersion.V1) {
+                String heartbeatNamespace = NamespaceService.getHeartbeatNamespace(
+                    pulsar().getAdvertisedAddress(), pulsar().getConfiguration());
+                topic = String.format("persistent://%s/healthcheck", heartbeatNamespace);
+            } else {
+                LOG.info("Using healthCheck with V2 topic name");
```

Comment:
I guess you can just add generic info log with topic name once you conclude topic eg:
`LOG.info("Doing healthCheck for {}", topic);`

Reply:
done